Build fails on Linux

When working on CodeSpaces, I got a build error on main (also reproes on WSL)

```text
/workspaces/Agents-for-net/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs(78,17): error CA1854: Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1854) [/workspaces/Agents-for-net/src/libraries/Authentication/Authentication.Msal/Microsoft.Agents.Authentication.Msal.csproj::TargetFramework=net8.0]
/workspaces/Agents-for-net/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs(78,17): error CA1854: Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1854) [/workspaces/Agents-for-net/src/libraries/Authentication/Authentication.Msal/Microsoft.Agents.Authentication.Msal.csproj::TargetFramework=.netstandard2.0]
/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs(95,17): error CA1854: Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1854) [/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/Microsoft.Agents.Builder.Dialogs.csproj::TargetFramework=net8.0]
/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs(20,54): error CA1859: Change type of field '_dialogs' from 'System.Collections.Generic.IDictionary<string, Microsoft.Agents.Builder.Dialogs.Dialog>' to 'System.Collections.Generic.Dictionary<string, Microsoft.Agents.Builder.Dialogs.Dialog>' for improved performance (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1859) [/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/Microsoft.Agents.Builder.Dialogs.csproj::TargetFramework=net8.0]
/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs(95,17): error CA1854: Prefer a 'TryGetValue' call over a Dictionary indexer access guarded by a 'ContainsKey' check to avoid double lookup (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1854) [/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/Microsoft.Agents.Builder.Dialogs.csproj::TargetFramework=.netstandard2.0]
/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs(20,54): error CA1859: Change type of field '_dialogs' from 'System.Collections.Generic.IDictionary<string, Microsoft.Agents.Builder.Dialogs.Dialog>' to 'System.Collections.Generic.Dictionary<string, Microsoft.Agents.Builder.Dialogs.Dialog>' for improved performance (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1859) [/workspaces/Agents-for-net/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/Microsoft.Agents.Builder.Dialogs.csproj::TargetFramework=.netstandard2.0]
/workspaces/Agents-for-net/src/libraries/Client/Microsoft.Agents.CopilotStudio.Client/Discovery/PowerPlatformEnvironment.cs(93,25): error CA1862: Prefer the string comparison method overload of 'string.Contains(string)' that takes a 'StringComparison' enum value to perform a case-insensitive comparison, but keep in mind that this might cause subtle changes in behavior, so make sure to conduct thorough testing after applying the suggestion, or if culturally sensitive comparison is not required, consider using 'StringComparison.OrdinalIgnoreCase' (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1862) [/workspaces/Agents-for-net/src/libraries/Client/Microsoft.Agents.CopilotStudio.Client/Microsoft.Agents.CopilotStudio.Client.csproj::TargetFramework=.netstandard2.0]
```

--- a/.github/workflows/ci-linux.yml
+++ b/.github/workflows/ci-linux.yml
@@ -1,4 +1,4 @@
-name: Local build and test
+name: Linux Local build and test 
 
 on:
   workflow_dispatch:

--- a/.github/workflows/ci-linux.yml
+++ b/.github/workflows/ci-linux.yml
@@ -1,0 +1,42 @@
+name: Local build and test
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.0.x        
+
+    - uses: dotnet/nbgv@master
+      with:
+          setAllVars: true        
+    - run: echo "NuGetPackageVersion $NuGetPackageVersion"      
+
+    - name: Restore
+      run: dotnet restore AgentSdk.proj
+  
+    - name: Build Debug
+      run: dotnet build --no-restore -c debug AgentSdk.proj
+
+    - name: Build Release
+      run: dotnet build --no-restore -c release AgentSdk.proj
+
+    - name: Package
+      run: dotnet pack --no-build --no-restore -c release src/Microsoft.Agents.SDK.sln 
+
+    - name: Test
+      run: dotnet test --no-build --no-restore -c debug ./src/ -f net8.0

--- a/.github/workflows/ci-linux.yml
+++ b/.github/workflows/ci-linux.yml
@@ -38,5 +38,5 @@ jobs:
     - name: Package
       run: dotnet pack --no-build --no-restore -c release src/Microsoft.Agents.SDK.sln 
 
-    - name: Test
+    - name: Test in net8 only
       run: dotnet test --no-build --no-restore -c debug ./src/ -f net8.0

--- a/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
+++ b/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
@@ -75,12 +75,12 @@ namespace Microsoft.Agents.Authentication.Msal
 
             // Get or create existing token. 
             _cacheList ??= new ConcurrentDictionary<Uri, ExecuteAuthenticationResults>();
-            if (_cacheList.TryGetValue(instanceUri, out ExecuteAuthenticationResults value))
+            if (_cacheList.TryGetValue(instanceUri, out ExecuteAuthenticationResults authResult))
             {
                 if (!forceRefresh)
                 {
-                    var accessToken = value.MsalAuthResult.AccessToken;
-                    var tokenExpiresOn = value.MsalAuthResult.ExpiresOn;
+                    var accessToken = authResult.MsalAuthResult.AccessToken;
+                    var tokenExpiresOn = authResult.MsalAuthResult.ExpiresOn;
                     if (tokenExpiresOn != null && tokenExpiresOn < DateTimeOffset.UtcNow.Subtract(TimeSpan.FromSeconds(30)))
                     {
                         accessToken = string.Empty; // flush the access token if it is about to expire.

--- a/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
+++ b/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
@@ -75,12 +75,12 @@ namespace Microsoft.Agents.Authentication.Msal
 
             // Get or create existing token. 
             _cacheList ??= new ConcurrentDictionary<Uri, ExecuteAuthenticationResults>();
-            if (_cacheList.ContainsKey(instanceUri))
+            if (_cacheList.TryGetValue(instanceUri, out ExecuteAuthenticationResults value))
             {
                 if (!forceRefresh)
                 {
-                    var accessToken = _cacheList[instanceUri].MsalAuthResult.AccessToken;
-                    var tokenExpiresOn = _cacheList[instanceUri].MsalAuthResult.ExpiresOn;
+                    var accessToken = value.MsalAuthResult.AccessToken;
+                    var tokenExpiresOn = value.MsalAuthResult.ExpiresOn;
                     if (tokenExpiresOn != null && tokenExpiresOn < DateTimeOffset.UtcNow.Subtract(TimeSpan.FromSeconds(30)))
                     {
                         accessToken = string.Empty; // flush the access token if it is about to expire.

--- a/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
+++ b/src/libraries/Authentication/Authentication.Msal/MsalAuth.cs
@@ -75,12 +75,12 @@ namespace Microsoft.Agents.Authentication.Msal
 
             // Get or create existing token. 
             _cacheList ??= new ConcurrentDictionary<Uri, ExecuteAuthenticationResults>();
-            if (_cacheList.TryGetValue(instanceUri, out ExecuteAuthenticationResults authResult))
+            if (_cacheList.TryGetValue(instanceUri, out ExecuteAuthenticationResults authResultFromCache))
             {
                 if (!forceRefresh)
                 {
-                    var accessToken = authResult.MsalAuthResult.AccessToken;
-                    var tokenExpiresOn = authResult.MsalAuthResult.ExpiresOn;
+                    var accessToken = authResultFromCache.MsalAuthResult.AccessToken;
+                    var tokenExpiresOn = authResultFromCache.MsalAuthResult.ExpiresOn;
                     if (tokenExpiresOn != null && tokenExpiresOn < DateTimeOffset.UtcNow.Subtract(TimeSpan.FromSeconds(30)))
                     {
                         accessToken = string.Empty; // flush the access token if it is about to expire.

--- a/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs
+++ b/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs
@@ -92,11 +92,11 @@ namespace Microsoft.Agents.Builder.Dialogs
 
             AssertionHelpers.ThrowIfNull(dialog, nameof(dialog));
 
-            if (_dialogs.TryGetValue(dialog.Id, out Dialog value))
+            if (_dialogs.TryGetValue(dialog.Id, out Dialog dialogValue))
             {
                 // If we are trying to add the same exact instance, it's not a name collision.
                 // No operation required since the instance is already in the dialog set.
-                if (value == dialog)
+                if (dialogValue == dialog)
                 {
                     return this;
                 }

--- a/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs
+++ b/src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs
@@ -17,7 +17,7 @@ namespace Microsoft.Agents.Builder.Dialogs
     /// </summary>
     public class DialogSet
     {
-        private readonly IDictionary<string, Dialog> _dialogs = new Dictionary<string, Dialog>();
+        private readonly Dictionary<string, Dialog> _dialogs = new Dictionary<string, Dialog>();
         private readonly DialogState _dialogState;
         private readonly IStatePropertyAccessor<DialogState> _dialogStateProperty;
         private string _version;
@@ -92,11 +92,11 @@ namespace Microsoft.Agents.Builder.Dialogs
 
             AssertionHelpers.ThrowIfNull(dialog, nameof(dialog));
 
-            if (_dialogs.ContainsKey(dialog.Id))
+            if (_dialogs.TryGetValue(dialog.Id, out Dialog value))
             {
                 // If we are trying to add the same exact instance, it's not a name collision.
                 // No operation required since the instance is already in the dialog set.
-                if (_dialogs[dialog.Id] == dialog)
+                if (value == dialog)
                 {
                     return this;
                 }

--- a/src/libraries/Client/Microsoft.Agents.CopilotStudio.Client/Discovery/PowerPlatformEnvironment.cs
+++ b/src/libraries/Client/Microsoft.Agents.CopilotStudio.Client/Discovery/PowerPlatformEnvironment.cs
@@ -1,4 +1,5 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
 // Licensed under the MIT License.
 
 using System;
@@ -89,7 +90,7 @@ namespace Microsoft.Agents.CopilotStudio.Client.Discovery
 #if !NETSTANDARD
                     if ( directConnectUrl.Contains("tenants/00000000-0000-0000-0000-000000000000", StringComparison.OrdinalIgnoreCase))
 #else
-                    if (directConnectUrl!.ToLower().Contains("tenants/00000000-0000-0000-0000-000000000000"))
+                    if (directConnectUrl!.IndexOf("tenants/00000000-0000-0000-0000-000000000000", StringComparison.CurrentCultureIgnoreCase) >= 0)
 #endif
                     {
                         // Direct connection cannot be used, ejecting and forcing the normal settings flow: 

